File: Data Structures & Algorithms/majority-element/submission-2.js
class Solution {
    /**
     * @param {number[]} nums
     * @return {number}
     */
    majorityElement(nums) {
        let count = 0;
        let curr = 0;

        for(let num of nums) {
            if(count === 0) curr = num;

            if(curr === num) ++count;
            else --count;
        }

        return curr;
    }
}

File: Data Structures & Algorithms/top-k-elements-in-list/submission-2.py
import heapq

class Solution:
    def topKFrequent(self, nums: List[int], k: int) -> List[int]:
        ans = []
        count, freq = {}, [[] for _ in range(len(nums) + 1)]

        for num in nums:
            count[num] = 1 + count[num] if num in count else 1

        for key, value in count.items():
            freq[value].append(key)

        
        for i in range(len(freq) - 1, -1, -1):
            for val in freq[i]:
                ans.append(val)
                if len(ans) == k:
                    return ans

        raise Exception("ERROR")

